ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -118,19 +119,23 @@ public void run() {
           singleRun.cancel(true);
           executor.shutdownNow();
           executor = getTimeoutHandlingExecutor();
+          err = true;
         } catch (ExecutionException e) {
           LOG.info("Exception during executing compaction", e);
+          err = true;
         } catch (InterruptedException ie) {
           // do not ignore interruption requests
           return;
+        } catch (Throwable t) {
+          err = true;
         }
 
         doPostLoopActions(System.currentTimeMillis() - startedAt);
 
         // If we didn't try to launch a job it either means there was no work to do or we got
-        // here as the result of a communication failure with the DB.  Either way we want to wait
+        // here as the result of an error like communication failure with the DB, schema failures etc.  Either way we want to wait
         // a bit before, otherwise we can start over the loop immediately.
-        if (!launchedJob && !stop.get()) {
+        if ((!launchedJob || err) && !stop.get()) {
```

Comment:
why do we need an extra `err` atr, can't we re-use `launchedJob`, set it to false by default and change to `true` on submission?

Reply:
Before https://github.com/apache/hive/pull/3916/files#diff-c12003e4d6f63a4f8e0aa57aaa808fbf6066789bf4426f6d31123e9399570fd3 change launched Job was used to decide weather thread should go to sleep of not, but that still wasnt suffice to avoid instant respawn and trying to make connection to HMS. This flag ensure that in case of error/Exception we will wait for a dedicated sleep time before again trying to make connection.
cc @veghlaci05

Reply:
Actually what @deniskuzZ says makes sense, by setting launchedJob to false by default, there's no need for the err flag.

Reply:
Yes, It does. I was reluctant to change the default hence added this. If we are okay I will change the default launchedJob value. Thanks @deniskuzZ @veghlaci05